src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -131,28 +174,65 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string paramete
                     this.logger?.LogDebug("Running PowerShell rule engine");
                     evaluations = evaluations.Concat(this.powerShellRuleEngine.AnalyzeTemplate(templateContext));
                 }
+               
+                // Recursively handle nested templates 
+                dynamic jsonTemplate = JsonConvert.DeserializeObject(template);
+                dynamic processedTemplateResources = templatejObject["resources"];
+                var processedTemplateResourcesWithLineNumbers = templateContext.OriginalTemplate["resources"];
 
-                // For each rule we don't want to report the same line more than once
-                // This is a temporal fix
-                var evalsToValidate = new List<IEvaluation>();
-                var evalsToNotValidate = new List<IEvaluation>();
-                foreach (var eval in evaluations)
+                for (int i = 0; i < processedTemplateResources.Count; i++)
                 {
-                    if (!eval.Passed && eval.Result != null)
-                    {
-                        evalsToValidate.Add(eval);
-                    }
-                    else
+                    var currentProcessedResource = processedTemplateResources[i];
+                    dynamic currentProcessedResourceWithLineNumbers = processedTemplateResourcesWithLineNumbers[i];
+
+                    if (currentProcessedResource.type == "Microsoft.Resources/deployments")
                     {
-                        evalsToNotValidate.Add(eval);
+                        var nestedTemplateWithLineNumbers = currentProcessedResourceWithLineNumbers.properties.template;
+                        dynamic populatedNestedTemplate = JsonConvert.DeserializeObject(JsonConvert.SerializeObject(nestedTemplateWithLineNumbers));
+                        // Get the offset
+                        int nextLineNumberOffset = (nestedTemplateWithLineNumbers as IJsonLineInfo).LineNumber + lineNumberOffset - 1; // off by one
+                        // Check whether scope is set to inner or outer
+                        var scope = currentProcessedResourceWithLineNumbers.properties.expressionEvaluationOptions?.scope;
+
+                        int startOfTemplate = (nestedTemplateWithLineNumbers as IJsonLineInfo).LineNumber;
+                        string jsonNestedTemplate = ExtractNestedTemplate(template, startOfTemplate);
```

Comment:
Since `currentProcessedResourceWithLineNumbers` already has the parsed JSON with the line numbering, can you do something like this to simplify it, instead of manually parsing JSON?

```cs
string nestedTemplateString = null;
var nestedTemplateJToken = currentProcessedResourceWithLineNumbers.InsensitiveToken("properties.template");
if (nestedTemplateJToken != null)
{
    var startIndex = (nestedTemplateJToken as IJsonLineInfo).LineNumber;
    var endIndex = (nestedTemplateJToken.Last as IJsonLineInfo).LineNumber; // Might need to use something other than .Last
    if (startIndex < endIndex)
    {
        nestedTemplateString = string.Join(Environment.NewLine, template.Split(Environment.NewLine)[startIndex..endIndex]);
    }
}
```

Reply:
Hmm... that might not fully work in a template not properly formatted.  I think it will work though if you also get `LinePosition` - if that's non-zero for the start or end, you'll need to trim them accordingly.

Reply:
Also gone since we abandoned this approach